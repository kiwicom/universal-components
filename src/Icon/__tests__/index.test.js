// @flow

import * as React from 'react';
import { render } from 'react-native-testing-library';
import snapshotDiff from 'snapshot-diff';

import { Icon } from '..';

describe('Icon', () => {
  const { getByProps, getByText } = render(
    <Icon name="check" size="large" color="red" />
  );

  it('should have correct icon character for icon name', () => {
    expect(getByText('V')).toBeDefined();
  });

  it('should have correct props', () => {
    expect(getByProps({ size: 'large' })).toBeDefined();
    expect(getByProps({ name: 'check' })).toBeDefined();
    expect(getByProps({ color: 'red' })).toBeDefined();
  });

  it('should match snapshot diff between regular and custom icon', () => {
    const regular = render(<Icon name="check" />);
    const custom = render(<Icon name="check" size="large" color="red" />);

    expect(
      snapshotDiff(regular, custom, { contextLines: 2 })
    ).toMatchSnapshot();
  });

  it('throws an error if icon name is invalid', () => {
    // $FlowExpectedError we are able to mock it in Jest
    console.error = jest.fn(); // eslint-disable-line
    // $FlowExpectedError we expect this will be a flow error
    expect(() => render(<Icon name="__invalid-name__" />)).toThrowError(
      'Icon with name "__invalid-name__" does not exist.'
    );
  });
});
