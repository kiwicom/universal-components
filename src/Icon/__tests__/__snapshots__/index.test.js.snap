// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should match snapshot diff between regular and custom icon 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -6,8 +6,8 @@
        },
        Object {
-         \\"fontSize\\": 24,
+         \\"fontSize\\": 32,
        },
        Object {
-         \\"color\\": \\"#46515e\\",
+         \\"color\\": \\"red\\",
        },
        undefined,"
`;
